packages/xo-server/docs/vm-sync-hook.md

```diff
@@ -0,0 +1,80 @@
+# VM Sync Hook
+
+> This feature is currently _unstable_ and might change or be removed in the future.
+>
+> Feedbacks are very welcome on the [project bugtracker](https://github.com/vatesfr/xen-orchestra/issues).
+
+Before snapshotting (with or without memory, ie checkpoint), XO can notify the VM via an HTTP request.
+
+A typical use case is to make sure the VM is in a consistent state during the snapshot process, for instance by making sure database writes are flushed to the disk.
+
+## Configuration
+
+The feature is opt-in via a tag on the VM: `xo:notify-on-snapshot`.
+
+By default, it will be an HTTP request on the port `1727`, on the first IP address reported by the VM.
+
+If the _VM Tools_ (i.e. management agent) are not installed on the VM or if you which to use another URL, you can specify this in the tag: `xo:notify-on-snapshot=<URL>`.
+
+## Specification
+
+XO will waits for the request to be answered before starting the snapshot, but will not wait longer than _1 minute_.
+
+If the request fails for any reason, XO will go ahead with snapshot immediately.
+
```

Comment:
I think the output of the call should be added to the backup log

Reply:
This feature is not directly related to backups, it will be applied for all snapshots.